src/main/kotlin/racingcar/ui/ResultView.kt

```diff
@@ -1,32 +1,30 @@
 package racingcar.ui
 
 import racingcar.domain.Cars
-import racingcar.service.Output
+import racingcar.service.dto.Output
 
 object ResultView {
 
     fun printResult(output: Output) {
         printStartMessage()
-        printStates(output)
+        printHistory(output)
         printWinners(output.cars)
     }
 
     private fun printStartMessage() {
         println("실행 결과")
     }
 
-    private fun printStates(output: Output) {
-        for (i in 1..output.numberOfAttempts) {
-            printState(cars = output.cars, round = i)
-            println()
-        }
-    }
-
-    private fun printState(cars: Cars, round: Int) {
-        cars.forEach { car ->
-            val position = "-".repeat(car.getPositionOf(round).value)
-            println("${car.name} : $position")
+    private fun printHistory(output: Output) {
+        val racingHistory = output.racingHistory
+        val result = StringBuilder()
+        racingHistory.forEach { history ->
+            history.forEach { (carName, position) ->
+                result.append("$carName : ${"-".repeat(position)}\n")
+            }
+            result.append("\n")
         }
+        println(result.toString())
     }
```

Comment:
```suggestion
    private fun printHistory(output: Output) {
    val formattedHistory = output.racingHistory.joinToString(separator = "\n\n") { roundHistory ->
        roundHistory.entries.joinToString(separator = "\n") { (carName, position) ->
            "$carName : ${"-".repeat(position)}"
        }
    }
    println(formattedHistory)
}

```

joinToString도 내부적으로 StringBuilder를 사용하기 때문이 이렇게도 표현해볼 수 있을 것 같네요!